### PR TITLE
update(Translate): Add key support. Make params and context optional.

### PR DESCRIPTION
**Pull request by milesj**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

- Adds unique key support through the `k` prop or `key` option.
- Updates params to support undefined and null.
- Updates all our usage to use keys.

## Motivation and Context

We're moving towards a direction in which phrases will require keys, so this is the first step.

## Testing

Locally.

## Screenshots

N/A

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by alecklandgraf** [comment]
I would lean against using `key` as it's special in React.

**Comment by milesj** [comment]
@alecklandgraf The prop is `k`. We only use `key` when using the function form.

File: packages/core/src/components/Autocomplete/index.tsx [hunk 1]
```diff
@@ -437,7 +437,15 @@ export default class Autocomplete<T extends Item> extends React.Component<Props<
...
       optional,
       placeholder:
-        placeholder || T.phrase('Search', {}, 'Search field within an autocomplete input field'),
+        placeholder ||
+        T.phrase(
+          'Search',
+          {},
+          {
+            context: 'Search field within an autocomplete input field',
+            key: 'lunar.common.search',
```
Comment (alecklandgraf) [inline packages/core/src/components/Autocomplete/index.tsx]: I see, only using `key` for `T.phrase`

File: packages/core/src/components/EmojiPicker/index.tsx [hunk 2]
```diff
@@ -103,36 +103,40 @@ export class EmojiPicker extends React.Component<Props & WithStylesProps> {
...
-      skinNone: T.phrase('No skin tone', {}, 'Emoji picker skin tone label'),
-      skinLight: T.phrase('Light skin tone', {}, 'Emoji picker skin tone label'),
-      skinMediumLight: T.phrase('Medium-light skin tone', {}, 'Emoji picker skin tone label'),
-      skinMedium: T.phrase('Medium skin tone', {}, 'Emoji picker skin tone label'),
-      skinMediumDark: T.phrase('Medium-dark skin tone', {}, 'Emoji picker skin tone label'),
-      skinDark: T.phrase('Dark skin tone', {}, 'Emoji picker skin tone label'),
-      search: T.phrase('Search emojis', {}, 'Emoji picker search input field label'),
-      searchA11y: T.phrase(
-        'Search for emojis by keyword',
+      recentlyUsed: T.phrase('Recently Used', {}, { key: 'lunar.emoji.recentlyUsed' }),
```
Comment (alecklandgraf) [inline packages/core/src/components/EmojiPicker/index.tsx]: can you keep the context here too? `{ key: 'lunar.emoji.recentlyUsed', context: 'Emoji picker commonly used feature type'}`

Comment (milesj) [reply packages/core/src/components/EmojiPicker/index.tsx]: I removed the context from the `EmojiPicker` as I feel like the phrase + key name is enough to infer it has to do with emojis.

**Pull request merged**